packages/components/src/components/assets/CoinLogo/CoinLogo.tsx

```diff
@@ -1,38 +1,130 @@
+            } catch (error) {
+                console.error('Error checking URL accessibility:', error);
+                setIsAccessible(false);
+            }
+        };
+
+        if (url) {
+            checkAccessibility();
+        }
+    }, [url]);
+
+    return isAccessible;
+};
+
+export const useAssetUrl = (coingeckoId: string, contractAddress?: string) => {
+    const fileName = contractAddress ? `${coingeckoId}_${contractAddress}` : coingeckoId;
+    const assetUrl = fileName ? `${UPDATED_ICONS_LIST_URL_BASE}${fileName}@2x.webp` : '';
+    const isAssetUrlAccessible = useCheckUrlAccessibility(assetUrl);
+    const iconUrl = fileName && isAssetUrlAccessible ? assetUrl : undefined;
+
+    return iconUrl;
+};
+
 export interface CoinLogoProps extends ImgHTMLAttributes<HTMLImageElement> {
-    symbol: CoinType;
+    symbol?: CoinType;
     className?: string;
     size?: number;
     index?: number;
+    coingeckoId?: string;
+    contractAddress?: string;
+    quality?: Quality;
 }
 
-const SvgWrapper = styled.div<Omit<CoinLogoProps, 'symbol'>>`
+const Wrapper = styled.div<{ $size: number }>`
     display: inline-block;
-    width: ${({ size }) => size}px;
-    height: ${({ size }) => size}px;
+    width: ${({ $size }) => $size}px;
+    height: ${({ $size }) => $size}px;
+    margin: ${spacingsPx.xxs};
 
     div {
-        width: ${({ size }) => size}px;
-        height: ${({ size }) => size}px;
-        line-height: ${({ size }) => size}px;
+        width: ${({ $size }) => $size}px;
+        height: ${({ $size }) => $size}px;
+        line-height: ${({ $size }) => $size}px;
     }
 `;
 
-export const CoinLogo = ({ symbol, className, size = 32, ...rest }: CoinLogoProps) => (
-    <SvgWrapper className={className} size={size} {...rest}>
-        <ReactSVG
-            src={COINS[symbol]}
-            beforeInjection={svg => {
-                svg.setAttribute('width', `${size}px`);
-                svg.setAttribute('height', `${size}px`);
-            }}
-            loading={() => <span className="loading" />}
-        />
-    </SvgWrapper>
+const StyledIconImage = styled.img<{ $size: number }>(
+    ({ $size }) => `
+        width: ${$size}px;
+        height: ${$size}px;
+    `,
+);
+
+type SVGCoinLogoProps = {
+    $symbol: CoinType;
+    $size?: number;
+};
+
+const SVGCoinLogo = ({ $symbol, $size = 32 }: SVGCoinLogoProps) => (
+    <ReactSVG
+        src={$symbol}
+        beforeInjection={svg => {
+            svg.setAttribute('width', `${$size}px`);
+            svg.setAttribute('height', `${$size}px`);
+        }}
+    />
 );
+
+export const CoinLogo = ({
+    symbol,
+    className,
+    size = 32,
+    coingeckoId = '',
+    contractAddress,
+    quality = 'medium',
+    ...rest
+}: CoinLogoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+
+    const iconUrl = useAssetUrl(coingeckoId, contractAddress);
+    const firstCharacter = coingeckoId && coingeckoId[0].toUpperCase();
```

Comment:
nit: 
```
    const firstCharacter = coingeckoId?.[0].toUpperCase();
```

Reply:
This change leads to errors (the picture shows already renamed component to AssetLogo):
![errrr](https://github.com/user-attachments/assets/24d4f8a1-e878-4773-b55e-25432951f11c)
 
 I'll leave this as it was before.

Reply:
so `coingeckoId[0]` is `undefined`. I might be missing something but I think this is also not resolved in your previous code.

maybe this way it could work?

```
    const firstCharacter = coingeckoId?.[0]?.toUpperCase();

Reply:
Fortunately this doesn't work. Not sure why. And in my previous code, it works for me. Mystery? Anyway this is such a nitpick that I'm not sure if we want to spend more time on this.

Reply:
No problem, let's skip it